src/Analyzers/CSharp/Tests/MatchFolderAndNamespace/CSharpMatchFolderAndNamespaceTests.cs

```diff
@@ -886,5 +887,70 @@ class P
 
             await testState.RunAsync();
         }
+
+        [Fact]
+        [WorkItem(58372, "https://github.com/dotnet/roslyn/issues/58372")]
+        public async Task InvalidProjectName_ChangeNamespace()
+        {
+            var defaultNamespace = "Invalid-Namespace";
```

Comment:
Do we have any tests for if this contains something like a comment but is otherwise valid (like System.Foo // look a comment!) or if it's empty?

Reply:
Empty, yes. Comments, no, though I would hope anything with a space would fail the IsIdentifier check, I can add one.